Makefile

```diff
@@ -12,46 +12,94 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-VERSION ?= v0.1.7
+ifeq ($(strip $(shell git status --porcelain 2>/dev/null)),)
+  GIT_TREE_STATE=clean
+else
+  GIT_TREE_STATE=dirty
+endif
+
+## Bump the version in the version file. Set BUMP to [ patch | major | minor ]
+BUMP := patch
+
+COMMIT ?= $(shell git rev-parse HEAD)
+BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD)
+VERSION ?= $(shell cat VERSION)
 REGISTRY ?= digitalocean
 
-all: clean ci compile build push
+LDFLAGS ?= -X github.com/digitalocean/digitalocean-cloud-controller-manager/vendor/k8s.io/kubernetes/pkg/version.gitVersion=$(VERSION) -X github.com/digitalocean/digitalocean-cloud-controller-manager/vendor/k8s.io/kubernetes/pkg/version.gitCommit=$(COMMIT) -X github.com/digitalocean/digitalocean-cloud-controller-manager/vendor/k8s.io/kubernetes/pkg/version.gitTreeState=$(GIT_TREE_STATE)
+PKG ?= github.com/digitalocean/digitalocean-cloud-controller-manager/cloud-controller-manager/cmd/digitalocean-cloud-controller-manager
+
+all: test
+
+publish: clean ci compile build push
+publish-dev: clean ci compile-dev build-dev push-dev
+
+ci: check-headers gofmt govet golint test
+
+bump-version: 
+	@go get -u github.com/jessfraz/junk/sembump # update sembump tool
+	$(eval NEW_VERSION = $(shell sembump --kind $(BUMP) $(VERSION)))
+	@echo "Bumping VERSION from $(VERSION) to $(NEW_VERSION)"
+	@echo $(NEW_VERSION) > VERSION
+	@cp releases/${VERSION}.yml releases/${NEW_VERSION}.yml
+	@sed -i'' -e 's/${VERSION}/${NEW_VERSION}/g' releases/${NEW_VERSION}.yml
+	@sed -i'' -e 's/${VERSION}/${NEW_VERSION}/g' docs/getting-started.md
+	@sed -i'' -e 's/${VERSION}/${NEW_VERSION}/g' README.md
+	@rm README.md-e docs/getting-started.md-e releases/${NEW_VERSION}.yml-e
 
-.PHONY: clean
 clean:
-	GOOS=linux go clean -i -x ./...
+	@echo "==> Cleaning releases"
+	@GOOS=linux go clean -i -x ./...
 
-.PHONY: compile
 compile:
-	CGO_ENABLED=0 GOOS=linux go build -ldflags '-X github.com/digitalocean/digitalocean-cloud-controller-manager/vendor/k8s.io/kubernetes/pkg/version.gitVersion=$(VERSION)' ./cloud-controller-manager/cmd/digitalocean-cloud-controller-manager
+	@echo "==> Building the project"
+	@CGO_ENABLED=0 GOOS=linux go build -ldflags "$(LDFLAGS)" ${PKG}
+
+compile-dev:
+	@echo "==> Building the project"
+	$(eval VERSION = dev)
+	@CGO_ENABLED=0 GOOS=linux go build -ldflags "$(LDFLAGS)" ${PKG}
 
-.PHONY: build
 build:
-	docker build -t $(REGISTRY)/digitalocean-cloud-controller-manager:$(VERSION) -f cloud-controller-manager/cmd/digitalocean-cloud-controller-manager/Dockerfile .
+	@echo "==> Building the docker image"
+	@docker build -t $(REGISTRY)/digitalocean-cloud-controller-manager:$(VERSION) -f cloud-controller-manager/cmd/digitalocean-cloud-controller-manager/Dockerfile .
+
+build-dev:
```

Comment:
(not blocking): I see the convenience in this, but not sure we need a `build-dev` and `push-dev` target. The same can be accomplished with `VERSION=dev make push`.

Reply:
Agree. Removed the `-dev` targets and changed the README description to `VERSION=dev make push`.